### PR TITLE
Connect shard throughput

### PR DESCRIPTION
**Pull request by fulmicoton**
Tested on simian with a pathologic workload:
```
- chico(rate=40mb)
- chico(rate=10kb)*10
```

On the left, without any per-shard weighting.
On the right, with per-shard weighting.

indexing throughput
image

cpu usage
image

### CODE REVIEW

File: quickwit/quickwit-control-plane/src/indexing_scheduler/mod.rs [hunk 1]
```diff
@@ -121,6 +121,28 @@ impl fmt::Debug for IndexingScheduler {
...
+/// It does not take in account the variation that could raise from the different
+/// doc mapping / nature of the data, etc.
+fn compute_load_per_shard(shard_entries: &[&ShardEntry]) -> NonZeroU32 {
+    let num_shards = shard_entries.len().max(1) as u32;
+    let average_throughput_per_shard: u32 = shard_entries
+        .iter()
+        .map(|shard_entry| u32::from(shard_entry.ingestion_rate.0))
+        .sum::<u32>()
+        .div_ceil(num_shards);
+    let num_cpu_millis = (PIPELINE_FULL_CAPACITY.cpu_millis() * average_throughput_per_shard) / 20;
```
[the remark this answers is not in the document]
Comment (fulmicoton) [reply quickwit/quickwit-control-plane/src/indexing_scheduler/mod.rs:141]: average_throughput_per_shard cannot exceed 5mb

File: quickwit/quickwit-control-plane/src/indexing_scheduler/mod.rs [hunk 2]
```diff
@@ -121,6 +121,28 @@ impl fmt::Debug for IndexingScheduler {
     }
 }
 
+/// Computes the CPU load associated to a single shard of a given index.
+///
+/// The array passed contains all of data we have about the shard of the index.
+/// This function averages their statistics.
+///
+/// For the moment, this function only takes in account the measured throughput,
+/// and assumes a constant CPU usage of 4 vCPU = 30mb/s.
```
Comment (fulmicoton) [inline quickwit/quickwit-control-plane/src/indexing_scheduler/mod.rs:130]:
```suggestion
/// and assumes a constant CPU usage of 4 vCPU = 20mb/s.
```

File: quickwit/quickwit-control-plane/src/indexing_scheduler/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (fulmicoton) [reply quickwit/quickwit-control-plane/src/indexing_scheduler/mod.rs:141]: I fixed the comment to 20mb/s

[the remark this answers is not in the document]
Comment (fulmicoton) [reply quickwit/quickwit-control-plane/src/indexing_scheduler/mod.rs]: I wrangled the code a little bit, and added a comment to make that clearer.

**Review by trinity-1686a — APPROVED** [review]
some lints are firing, otherwise lgtm

**Comment by fulmicoton** [comment]
Closed and reopened as #5196

**Pull request closed without merging**